ui/src/data/source.tsx

```diff
@@ -531,203 +500,190 @@ function isInternalAttribute(attribute: string): boolean {
   return attribute.startsWith("t_");
 }
 
-function makePathAttribute(attribute: string) {
-  return "t_path_" + attribute;
-}
-
-function makeNumChildrenAttribute(attribute: string) {
-  return "t_numChildren_" + attribute;
-}
+function literalFromDataValue(value: DataValue): tanagra.LiteralV2 {
+  let dataType = tanagra.DataTypeV2.Int64;
+  if (typeof value === "string") {
+    dataType = tanagra.DataTypeV2.String;
+  } else if (typeof value === "boolean") {
+    dataType = tanagra.DataTypeV2.Boolean;
+  }
 
-function attributeValueFromDataValue(value: DataValue): tanagra.AttributeValue {
   return {
-    int64Val: typeof value === "number" ? value : undefined,
-    stringVal: typeof value === "string" ? value : undefined,
-    boolVal: typeof value === "boolean" ? value : undefined,
+    dataType,
+    valueUnion: {
+      int64Val: typeof value === "number" ? value : undefined,
+      stringVal: typeof value === "string" ? value : undefined,
+      boolVal: typeof value === "boolean" ? value : undefined,
+    },
   };
 }
 
-function dataValueFromAttributeValue(
-  value?: tanagra.AttributeValue | null
-): DataValue {
-  return value?.int64Val ?? value?.stringVal ?? value?.boolVal ?? -1;
+function dataValueFromLiteral(value?: tanagra.LiteralV2 | null): DataValue {
+  switch (value?.dataType) {
```

Comment:
Do you also need to handle dates?

Reply:
Yes, thanks. I had forgotten you added them.